rfcs/config-luaurc.md

```diff
@@ -0,0 +1,70 @@
+# Configure analysis via .luaurc
+
+## Summary
+
+Introduces a way to configure type checker and linter using JSON-like .luaurc files
+
+## Motivation
+
+While Luau analysis tools try to provide sensible defaults, it's difficult to establish the rules that work for all code.
+For example, some packages may decide that unused variables aren't interesting; other packages may decide that all files should be using strict typechecking mode.
+
+While it's possible to configure some aspects of analysis behavior using --! comments, it can be cumbersome to replicate this in all files.
+
+## Design
+
+To solve this problem, we are going to introduce support for `.luaurc` files for users of command-line Luau tools.
+For a given .lua file, Luau will search for .luaurc files starting from the folder that the .lua file is in; all files in the ancestry chain will be parsed and their configuration
+applied. When multiple files are used, the file closer to the .lua file overrides the settings.
+
+.luaurc is a JSON file that can also contain comments and trailing commas. The file can have the following keys:
+
```

Comment:
Is it worth having an "entrypoints" list which says which files are considered entry points to the API?

Reply:
Ditto search path configuration?

Reply:
Do you mean for package configuration? I think it's too early to specify this right now, even though I definitely agree that we'd expect to have this in this file in the future. E.g. in the future I could expect something like "dependencies" that has a module name -> path mapping for require as part of this file.